test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_trcomp.py

```diff
@@ -208,5 +208,292 @@ def test_trcomp_debug(self, patched, ecr_image, sagemaker_session, tmpdir, py_ve
         xla_metrics_file = os.path.join(tmpdir, 'compiler', 'XLA_METRICS_FILE.txt')
         assert os.path.exists(xla_metrics_file)
 
+@pytest.mark.integration("sagmaker-training-compiler")
+@pytest.mark.processor("gpu")
+@pytest.mark.skip_py2_containers
+@pytest.mark.skip_huggingface_containers
+@pytest.mark.skip_cpu
+@mock.patch('sagemaker.huggingface.TrainingCompilerConfig.validate', return_value=None)
+class TestSingleNodeMultiGPU:
+    '''
+    All Single Node Single GPU tests go here.
+    '''
+    @pytest.mark.model("bert-large")
+    def test_trcomp_default(self, patched, ecr_image, sagemaker_session, tmpdir, py_version, capsys):
+        '''
+        Tests the default configuration of SM trcomp
+        '''
+        transformers_version = get_transformers_version(ecr_image)
+        git_config = {'repo': 'https://github.com/huggingface/transformers.git', 'branch': 'v'+transformers_version}
+
+        instance_count = 1
+        instance_type = "ml.p3.16xlarge"
+        
+        source_dir = (
+            "./examples/question-answering"
+            if Version(transformers_version) < Version("4.6")
+            else "./examples/pytorch/question-answering"
+        )
+        
+        with timeout(minutes=DEFAULT_TIMEOUT):
+            estimator = HuggingFace(
+                compiler_config=TrainingCompilerConfig(),
+                entry_point='run_qa.py',
+                source_dir=source_dir,
+                git_config=git_config,
+                metric_definitions=metric_definitions,
+                role='SageMakerRole',
+                image_uri=ecr_image,
+                instance_count=instance_count,
+                instance_type=instance_type,
+                sagemaker_session=sagemaker_session,
+                hyperparameters=hyperparameters,
```

Comment:
The hyperparameter `max_steps` needs to be multiplied by number of GPUs and instances to get a meaningful distribution